### PR TITLE
Fix #32: checks and revision of manifest files

### LINKED ISSUE #32: Improve OLM Integration
The package naming is not in sync with the OLM specification: https://operator-sdk.netlify.app/docs/olm-integration/user-guide/#creating-a-bundle

### PR DESCRIPTION
**Pull request by ricardozanini**
Signed-off-by: Ricardo Zanini <ricardozanini@gmail.com>

See: #32 

A lot of small issues addressed to properly install on OLM and update OperatorHub. See this doc for reference: https://operator-sdk.netlify.app/docs/olm-integration/

### CODE REVIEW

File: deploy/crds/apps.m88i.io_v1alpha1_nexus_cr.yaml [hunk 1]
```diff
@@ -20,9 +20,9 @@ spec:
   # Data persistence details
   persistence:
     # Should we persist Nexus data? Yes, please. (turn this to false only if you're evaluating this resource)
-    persistent: true
+    persistent: false
```
Comment (LCaparelli) [inline deploy/crds/apps.m88i.io_v1alpha1_nexus_cr.yaml:23]: Should this be set to false here? If so, we might want to change the comment above it (specifically the "Yes, please" part).

File: deploy/olm-catalog/nexus-operator/0.2.0/nexus-operator.v0.2.0.clusterserviceversion.yaml [hunk 2]
```diff
@@ -177,7 +88,7 @@ spec:
     After installing it, you will have to grab the `admin` user password from the deployed container. There's a file auto generated in `/nexus-data/admin.password`.
     Use `cat` to read the file and view the password. Use it to login for the first time and follow the on screen instructions to have the Nexus server ready for use.
 
-    If you experience any issues or have any ideas for new features, please [file an issue in our Github repository](https://github.com/m88i/nexus-operator/issues).
+    If you experience any issues or have any ideas for new features, please [file an issue in our Github repository](https://github.com/m88i/nexus-operator/issues) or send an email to our maillist: nexus-operator@googlegroups.com
```
Comment (LCaparelli) [inline deploy/olm-catalog/nexus-operator/0.2.0/nexus-operator.v0.2.0.clusterserviceversion.yaml:91]: How about we make it a mailto: link? \[nexus-operator@googlegroups.com\]\(mailto:nexus-operator@googlegroups.com\)

File: build/bin/entrypoint [hunk 3]
```diff
@@ -1,12 +1,3 @@
 #!/bin/sh -e
 
-# This is documented here:
```
Comment (LCaparelli) [inline build/bin/entrypoint:3]: I'm not sure I follow, why is this no longer necessary?

File: hack/go-build.sh [hunk 4]
```diff
@@ -23,7 +23,7 @@ REPO=https://github.com/m88i/nexus-operator
 BRANCH=master
 REGISTRY=quay.io/m88i
 IMAGE=nexus-operator
-TAG=0.2.0-rc1
+TAG=${OP_VERSION}
```
Comment (LCaparelli) [inline hack/go-build.sh:26]: Where is this variable coming from?

Comment (LCaparelli) [reply hack/go-build.sh:26]: I have a feeling we're missing a `source hack/export-version.sh` here :-)

File: pkg/apis/addtoscheme_apps_v1alpha1.go [hunk 5]
```diff
@@ -19,11 +19,9 @@ package apis
...
 import (
 	"github.com/m88i/nexus-operator/pkg/apis/apps/v1alpha1"
-	routev1 "github.com/openshift/api/route/v1"
-	ingressv1beta1 "k8s.io/api/networking/v1beta1"
 )
 
 func init() {
 	// Register the types with the Scheme so the components can map objects to GroupVersionKinds and back
-	AddToSchemes = append(AddToSchemes, v1alpha1.SchemeBuilder.AddToScheme, routev1.Install, ingressv1beta1.AddToScheme)
+	AddToSchemes = append(AddToSchemes, v1alpha1.SchemeBuilder.AddToScheme)
```
Comment (LCaparelli) [inline pkg/apis/addtoscheme_apps_v1alpha1.go:26]: What's the story here, what made those no longer necessary?

Comment (ricardozanini) [reply pkg/apis/addtoscheme_apps_v1alpha1.go:26]: Now we control this schemas in the `pkg/framework/controller_watcher.go`. This file is auto-generated, so we shouldn't being adding things in there. xD

Comment (LCaparelli) [reply pkg/apis/addtoscheme_apps_v1alpha1.go:26]: Gotcha hehe

**Pull request merged**